xwiki-platform-core/xwiki-platform-ckeditor/xwiki-platform-ckeditor-test/xwiki-platform-ckeditor-test-pageobjects/src/main/java/org/xwiki/ckeditor/test/po/MacroDialogEditModal.java

```diff
@@ -46,6 +47,7 @@ public MacroDialogEditModal waitUntilReady()
      */
     public void clickSubmit()
     {
-        getDriver().findElement(By.cssSelector(".gadget-editor-modal .modal-footer .btn-primary")).click();
+        getDriver().findElement(
+                By.xpath("//*[contains(@class, 'editor-modal')]//*[contains(@class, 'modal-footer')]//*[contains(@class, 'btn-primary')]")).click();
```

Comment:
Same, the CSS selector might be simpler. XPath is useful when you need to match the content of an HTML element, or its parent or a sibling.